# Análise de Previsão de Churn


In [1]:
# Montar o drive para que o nosso script consiga acessar a base de dados.
from google.colab import drive
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=email%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdocs.test%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive.photos.readonly%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fpeopleapi.readonly&response_type=code

Enter your authorization code:
··········
Mounted at /content/drive


In [0]:
import os
os.chdir("/content/drive/My Drive")  #MAPEAR PARA O LOCAL NO DRIVE ONDE SE ENCONTRA ESSE ARQUIVO

In [3]:
!ls #Ver a localização desse script nas pastas. o ARQUIVO DE CHURN DEVE ESTAR LISTADO ABAIXO

 20150619_ReuniaoAcompanhamento06.docx
 20150820_ReuniaoAcompanhamento08.docx
 2016_IPVA_boleto.pdf
 2017.101.GW.zip
'2018-07-25 13:12:40_RGB_Video_.avi'
'2018-08-15 10-07-11.mp4'
 20180822_095218.jpg
 20180906_113839.jpg
 20180911_110614.jpg
 20180917_100509.jpg
 635895851822503045sched_.xlsx
'Adsense Mãe.gsheet'
'AI (1).rar'
 AI.rar
 Aniversário
'Apresentação sem título.gslides'
'aranha social fluxograma.gdraw'
 Aula04.rar
'Aula Clusterização e Reconhecimento de padrões.pptx'
'aula data mining.gsheet'
'Aulas ICA'
 BI
 BigOil
'BIG OIL'
'bi-master - programa e quiz.gdoc'
 breastCancer.csv
'bug restart (1).zip'
'bug restart.zip'
 Churn.csv
 Churn.ipynb
 Classroom
'Colab Notebooks'
 Colômbia.gdoc
'Comprovante de compra.pdf'
 consumer_complaints.csv
'Controle de pagamento 2017_MAR.xlsx.gsheet'
'Cópia de employee-timesheet.gsheet'
'Cópia de Monthly Employee Timesheet by Hours with Calculations (1).gsheet'
'Cópia de Monthly Employee Timesheet by Hours with Calculations.gsheet'
'

Neste script iremos prever a espécie de uma planta usando o algoritmo KNN (K Nearest Neighbor).

# Setup

##Importar bibliotecas necessárias

In [0]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score

%matplotlib inline

# Exercício 1

Carregue o arquivo csv chamado 'Iris.csv' usando pandas e salve a base em uma variável chamada df. (dica: pd.read_csv('<nome_da_base.csv>')) Imprima as primeiras 10 linhas do dataframe e verifique se tudo foi carregado corretamente. (dica: df.head())

In [0]:
df = pd.read_csv('Iris.csv')

In [6]:
df.head()

SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0            5.1           3.5            1.4           0.2  Iris-setosa
1            4.9           3.0            1.4           0.2  Iris-setosa
2            4.7           3.2            1.3           0.2  Iris-setosa
3            4.6           3.1            1.5           0.2  Iris-setosa
4            5.0           3.6            1.4           0.2  Iris-setosa

# Exercício 2: 

Que coluna indica a espécie da planta? Quantas plantas existem na sua base de dados? (dica: len(?))


In [7]:
#coluna rótulo: 'Species'
# número de plantas:
len(df)

150

# Exercício 3:

Use df.describe() para explorar cada coluna. Todos devem ser igual a 150 pois não temos missing values!!

In [8]:
df.describe()

SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
count     150.000000    150.000000     150.000000    150.000000
mean        5.843333      3.054000       3.758667      1.198667
std         0.828066      0.433594       1.764420      0.763161
min         4.300000      2.000000       1.000000      0.100000
25%         5.100000      2.800000       1.600000      0.300000
50%         5.800000      3.000000       4.350000      1.300000
75%         6.400000      3.300000       5.100000      1.800000
max         7.900000      4.400000       6.900000      2.500000

# Exercício 5:

Verifique o número de colunas da base de dados (dica: df.shape)

Separe a base de dados em atributos de entrada e rótulos. Associe os atributos a variável X e os rótulos a variável y.

Verifique as primeiras linhas das bases separadas.

In [9]:
df.shape

(150, 5)

In [0]:
X = df.iloc[:,0:4]
y = df.iloc[:,4]

In [11]:
X.head()

SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
0            5.1           3.5            1.4           0.2
1            4.9           3.0            1.4           0.2
2            4.7           3.2            1.3           0.2
3            4.6           3.1            1.5           0.2
4            5.0           3.6            1.4           0.2

In [12]:
y.head()

0    Iris-setosa
1    Iris-setosa
2    Iris-setosa
3    Iris-setosa
4    Iris-setosa
Name: Species, dtype: object

# Exercício 6: 

Separa a base de treino (70%) e teste (30%) e associe-as as variáveis X_train, X_test, y_train, y_test (dica: train_test_split(X,y, test_size=0.3))

In [0]:
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.3)

# Exercício 7: 

Crie um classificador KNN com 5 vizinhos. (hint: KNeighborsClassifier(n_neighbors=5))

In [0]:
knn = KNeighborsClassifier(n_neighbors=5)

# Exercício 8:

Apresente a base de treino para ajustar o modelo. (hint: knn.fit(<base_de_treino_atributos>, <base_de_treino_rotulos>))

In [0]:
knn = knn.fit(X_train, y_train)

# Exercício 9:

Use o modelo para fazer previsões na base de teste. Associe os resultados de previsão a variável y_pred

In [0]:
y_pred = knn.predict(X_test)

# Exercício 10:

Determine quão acurado é o seu modelo. (dica: accuracy_score(<rotulo_real>, <rotulo_previsto>))

In [17]:
 accuracy_score(y_test,y_pred)

1.0